Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -109,7 +109,24 @@ public class BSONDecoder {
     }
 
     /// Initializes `self`.
-    public init() {}
+    public init(options: CodingStrategyOptions? = nil) {
+        self.applyOptions(options: options)
+    }
+
+    /// Initializes `self` by copying the options of another `BSONDecoder`.
+    internal init(copies other: BSONDecoder, options: CodingStrategyOptions?) {
+        self.userInfo = other.userInfo
+        self.dateDecodingStrategy = other.dateDecodingStrategy
+        self.uuidDecodingStrategy = other.uuidDecodingStrategy
+        self.dataDecodingStrategy = other.dataDecodingStrategy
+        self.applyOptions(options: options)
+    }
+
+    internal func applyOptions(options: CodingStrategyOptions?) {
```

Comment:
nit: generally `apply` works in the other direction in my experience ("apply the options represented here TO something else"), this seems to be "accepting" or "receiving" the options

Reply:
How does `configureWithOptions` sound?